### PR TITLE
Prevent model thrashing from unset num_ctx

### PR DESCRIPTION
**Pull request by rick-github**
TLDR: a model shouldn't be evicted due to num_ctx change if the client doesn't care about context size.

Client A loads a model with a context window different to the default or the value configured in the Modelfile:
```console
$ curl localhost:11434/api/generate -d '{"model":"llama3.2","options":{"num_ctx":65536}}'
$ ollama ps
NAME               ID              SIZE     PROCESSOR    UNTIL   
llama3.2:latest    a80c4f17acd5    13 GB    100% GPU     Forever 
```
Client B does a completion but doesn't specify a context window, causing the default value of 2048 to be used, resulting in eviction and immediate reload of the model.
```console
$ curl localhost:11434/api/generate -d '{"model":"llama3.2"}'
$ ollama ps
NAME               ID              SIZE      PROCESSOR    UNTIL   
llama3.2:latest    a80c4f17acd5    3.1 GB    100% GPU     Forever    
```
Client A sends another completion with the large context causing another eviction and reload.
```console
$ curl localhost:11434/api/generate -d '{"model":"llama3.2","options":{"num_ctx":65536}}'
$ ollama ps
NAME               ID              SIZE     PROCESSOR    UNTIL   
llama3.2:latest    a80c4f17acd5    13 GB    100% GPU     Forever    
```
If client B is not concerned about the context window, it shouldn't cause the eviction of an an already loaded model.  This is particularly noticeable when sharing a model between ollama and OpenAI endpoints - since the OpenAI endpoint can't set a context window, a model loaded via the ollama endpoint with a custom context window gets evicted by the next OpenAI request.

Thrashing can also occur when a client makes secondary completions after a primary completion, eg open-webui's auto-complete feature (see https://github.com/ollama/ollama/issues/7919#issuecomment-2560465774), or when a model is used for both completion and embedding (https://github.com/ollama/ollama/issues/6148#issuecomment-2568402497).

### CODE REVIEW

**Comment by richtong** [comment]
Just wondering is there a reason this isn't merged. Looks like it just needs a review? Or is it fixed elsewhere. The issue is closed but this looks still open. Affects openwebui performance quite a bit with large context local models. Thanks!  Great project!
